### DISCUSSION TITLE
how do I create a nvim plugin that uses fennel?

### DISCUSSION THREAD

**Discussion by ExpandingMan** [rktjmp/hotpot.nvim#111] [Q&A] [answered]
I'd like to create a brand new plugin using fennel instead of lua.  Is there a guide on how to do this?  I'd prefer to use hotpot since I'm already using it for my setup.  I can probably figure out how to transcompile plugin fennel code to lua following [the example of hotpot itself](https://github.com/rktjmp/hotpot.nvim/blob/master/lua/hotpot.lua), but I wanted to know if there is already a clean, straightforward option.  Thanks!

**Comment by ExpandingMan** [comment]
Oh, sorry, I jsut found [this](https://github.com/rktjmp/hotpot.nvim/blob/master/WRITING-PLUGINS.md) though it said it is out of date.

**Comment by ExpandingMan** [comment]
I think I more-or-less got this working.  However, I would really appreciate an explicit example of exactly what you'd expect to see in a plugin for using fennel if you'd be so kind :slightly_smiling_face:

**Comment by ExpandingMan** [comment]
Ah, this example fully answers my question: https://github.com/rktjmp/paperplanes.nvim

**Comment by rktjmp** [comment]
Sorry, I have some long-time-coming work in the pipe that will make this a bit nicer.

Basically you write a .hotpot.lua file that wraps the api.make stuff as seen in paperplanes and rebuild-on-save function (sort of like the fnl-init example) for you.

You'll hit a speed bump with hotpots lua precedence preference which will need you to turn on colocation, *but* be aware I am actually planning to deprecate that option I think as having it *and* the .hotpot file is confusing. So you'll hit a bit of code churn :frowning_face: 

I am in the midst of rebuilding some drives after a failure so it might take me a bit to finish out this stuff and update the documentation. I am hoping to get it all out this weekend.

**Comment by rktjmp** [comment, marked as answer]
I have update the docs to the latest method which should be quite painless. Unfortunately the previous make api could not be kept around for a deprecation period, but migrating to the new api should be simple, it's basically the same idea but simpler.

https://github.com/rktjmp/hotpot.nvim/blob/master/COOKBOOK.md#writing-a-plugin

**Comment by ExpandingMan** [comment]
Great, thanks!
